rudder-web/src/main/webapp/javascript/rudder/rudder-datatable.js

```diff
@@ -50,13 +50,45 @@ $.fn.dataTable.ext.order['dom-checkbox'] = function  ( settings, col )
 }
 
 /*
- * This function is used to resort a table after its sorting datas were changed ( like sorting function below)
+ * This function is used to resort a table after its sorting data were changed ( like sorting function below)
  */
 function resortTable (tableId) {
   var table = $("#"+tableId).DataTable({"retrieve" : true});
   table.draw();
 }
 
+
+$.fn.dataTable.ext.search.push(
+    function(settings, data, dataIndex ) {
+        //! param needs to be JSON only so we remove the hash tag # at index 0
+
+        var param = decodeURIComponent(window.location.hash.substring(1));
+        if (param !== "" && window.location.pathname === contextPath + "/secure/nodeManager/nodes") {
+            var obj = JSON.parse(param);
+            var min = obj.complianceFilter.min;
+            var max = obj.complianceFilter.max;
+            if (isNaN(min) || isNaN(max))
+                return true;
+
+            if (nodeCompliances !== undefined) {
+
+                //! here, we get the id of the row element by looking deep inside settings...
+                //! maybe there exists something cleaner.
+
+                var complianceArray = nodeCompliances[settings.aoData[dataIndex]._aData.id];
+                 if (complianceArray !== undefined) {
+                     var compliance = computeCompliancePercent(complianceArray);
+                     return compliance >= min && compliance <= max;
```

Comment:
In fact I think it should be strictly less than max

I think max can be undefined so you only check if compliance >= min 

and Perfect should not define a value for max and should be { min : 100 }